src/cargo/util/toml.rs

```diff
@@ -535,6 +535,10 @@ impl TomlManifest {
             debug!("manifest has no build targets");
         }
 
+        if let Err(e) = unique_build_targets(&targets) {
+            debug!("duplicate build target found: {}", e);
```

Comment:
I think you may want to change this to `bail!` to ensure the error gets propagated upwards